tests/test_state_user.py

```
@@ -74,3 +75,172 @@ def patched_exec(query):
+                if "localuser" in query_str.lower():
+                    result = MagicMock()
+                    result.one_or_none.return_value = None
+                    return result
+                return original_exec(query)
+
+            session.exec = patched_exec
+            password, error = state.add_user({"username": "newuser", "email": "taken@test.com", "role": "viewer"})
+            assert password is None
+            assert "Email" in error
+
+    def test_add_user_admin_role_rejected(self, patch_rx_session, seeded_db):
```

Ruff (0.15.5)

[warning] Unused method argument: `seeded_db`

(ARG002)